src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -98,7 +127,10 @@ public static String showTrace() {
      */
     private static boolean doSpin(final int numOfSpins) {
         boolean result = false;
-
+        if (dev == null) {
+            appendTrace("doSpin : Error, device is null");
+            return false;
+        }
```

Comment:
To keep with Max's single return, remove the return statements and make this if and the next if a waterfall if/else if/else statement

Reply:
Alright, that's a good idea. Added as a TODO in the bugfixes branch on my local repository.